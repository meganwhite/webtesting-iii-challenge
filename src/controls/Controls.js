import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div data-testid='controls' className="controls panel">
      <button data-testid='controlsButton1' disabled={!closed} onClick={toggleLocked} className="toggle-btn">
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button data-testid='controlsButton2' disabled={locked} onClick={toggleClosed} className="toggle-btn">
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
