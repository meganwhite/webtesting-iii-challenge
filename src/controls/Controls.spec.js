// Test away!
import React from 'react';
import {render, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";


import Controls from "./Controls"

describe("<Controls />", () => {
    it('should disable lock button if gate is open', () => {
        //render copy of control component
        const wrapper = render(<Controls />);
        const lockButton = wrapper.getByText("Lock Gate")
        //look for property on control component
        expect(lockButton).toBeDisabled();
    })

    it('should disable open button if gate is closed and locked', () => {
        //when it renders Controls, it will render in this state
        const {getByText} = render(<Controls locked={true} closed={true}/>)
        //target the button
        const openButton = getByText(/open gate/i);
        expect(openButton.disabled).toBe(true);

    })

    it('should not call lock function if gate is open', () => {
        const mockLock = jest.fn();
        const {getByText} = render(<Controls />);
        const lockButton = getByText(/lock gate/i);
        //look to see if function fired
        fireEvent.click(lockButton);
        expect(mockLock).not.toBeCalled();
    })

})