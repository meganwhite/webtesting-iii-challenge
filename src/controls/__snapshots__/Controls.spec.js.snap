// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> matches snapshot 1`] = `
<div
  className="controls panel"
  data-testid="controls"
>
  <button
    className="toggle-btn"
    data-testid="controlsButton1"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="controlsButton2"
  >
    Close Gate
  </button>
</div>
`;
